net/http/authz/authorizer.go

```diff
@@ -0,0 +1,96 @@
+package authz
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/golang/protobuf/proto"
+
+	"chain/database/raft"
+	"chain/errors"
+	"chain/net/http/authn"
+)
+
+const grantPrefix = "/core/grant/"
+
+type Authorizer struct {
+	raftDB        *raft.Service
+	raftPrefix    string
+	policyByRoute map[string][]string
+}
+
+func NewAuthorizer(rdb *raft.Service, prefix string, policyMap map[string][]string) *Authorizer {
+	return &Authorizer{
+		raftDB:        rdb,
+		raftPrefix:    prefix,
+		policyByRoute: policyMap,
+	}
+}
+
+func (a *Authorizer) Authorize(req *http.Request) error {
+	policies := a.policyByRoute[req.RequestURI]
+	if policies == nil {
+		return errors.New("missing policy on this route")
```

Comment:
I think this should be a CH000. Prob this package should export `ErrNotAuthorized` (replacing `errNotAuthorized` from chain/core) that core can put in the errors table as a 401, but any other errors would show up as 500s.